pvlib/irradiance.py

```diff
@@ -3117,3 +3117,72 @@ def complete_irradiance(solar_zenith,
                                      'dhi': dhi,
                                      'dni': dni})
     return component_sum_df
+
+
+def pvl_louche(ghi, solar_zenith, datetime_or_doy):
+    """
+    Determine DNI and GHI from GHI using Louche model.
+
+    Parameters
+    ----------
+    ghi : Series
```

Comment:
I think `ghi` and `solar_zenith` can be scalars and arrays as well as Series, so let's change the type to `numeric` for `ghi` and `solar_zenith`.